app/models/bitflyer_connection.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class BitflyerConnection < ApplicationRecord
+  SUPPORTED_CURRENCIES = ["BAT", "USD", "BTC", "ETH"].freeze
+
+  belongs_to :publisher
+
+  validates :recipient_id, uniqueness: true, allow_blank: true
+
+  attr_encrypted :access_token, :refresh_token, key: :encryption_key
+
+  after_save :update_default_currency, if: -> { saved_change_to_default_currency? }
+
+  validates :default_currency, inclusion: { in: SUPPORTED_CURRENCIES }, allow_nil: true
+
+  after_destroy :selected_wallet_provider
+
+  def prepare_state_token!
+    update(state_token: SecureRandom.hex(64).to_s)
+  end
+
+  def payable?
+    true
+  end
+
+  def japanese_account?
+    country&.upcase == 'JP'
+  end
+
+  def verify_url
+    ""
+  end
+
+  # Public: All the support currency pairs for BAT on the Bitflyer Exchange
+  # https://bitflyer.com/en-us/api
+  #
+  # Returns an array of currencies.
+  def supported_currencies
+    SUPPORTED_CURRENCIES
+  end
+
+  def access_token_expired?
+    access_expiration_time.present? && Time.now > access_expiration_time
+  end
+
+  # Makes a request to the Bitflyer API to refresh the current access_token
+  def refresh_authorization!
+    # Ensure we have an refresh_token.
+    return if refresh_token.blank?
+
+    authorization = Bitflyer::Auth.refresh(token: refresh_token)
+
+    # Update with the latest Authorization
+    update!(
+      access_token: authorization.access_token,
+      refresh_token: authorization.refresh_token,
+      expires_in: authorization.expires_in,
+      access_expiration_time: authorization.expires_in.seconds.from_now
+    )
+    # Reload the model so consumers will have the most up to date information.
+    reload
+  end
+
+  def sync_connection!
```

Comment:
Where is this called?

Reply:
I'll leave this in the code for now, we may need to implement a refresh flow at some point, but not needed for a staging test